tracer.go

```diff
@@ -80,7 +82,13 @@ func NewTracer(
 		reporter:    reporter,
 		injectors:   make(map[interface{}]Injector),
 		extractors:  make(map[interface{}]Extractor),
-		metrics:     *NewNullMetrics(),
+		headerKeys: HeadersConfig{
```

Comment:
I'd prefer if we set the custom values first and then apply the defaults after. This way if the user were to input empty strings for everything, we can fallback on the defaults.